From fb57880518f67a4585befed434e516cd9a88cf5b Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Mon, 18 Oct 2021 10:16:48 +0300
Subject: [PATCH] [SW-55053] hl: use reset causes defined in specs

reset causes enum was exported to specs to be
shared between the different projects, therefore
it can be included from header in specs for gaudi2
and renamed accordingly elsewhere in the code.

Change-Id: I35572c0dbfe09953aa58741dc3868c9301a80792
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 1ca68b2..ee1ed86 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -112,8 +112,8 @@
 
 	rst_src = hl_rst_src_get();
 	switch (rst_src) {
-	case HL_COLD_RST_SRC:
-	case HL_PRSTN_RST_SRC:
+	case HL_COLD_RST:
+	case HL_PRSTN_RST:
 		/* PCIe is reset, so must initialize PCIe */
 		preboot_init_pcie();
 		break;
@@ -134,19 +134,19 @@
 	char *cause;
 
 	switch (rst_src) {
-	case HL_COLD_RST_SRC:
+	case HL_COLD_RST:
 		cause = "cold";
 		break;
-	case HL_MANUAL_RST_SRC:
+	case HL_MANUAL_RST:
 		cause = "manual";
 		break;
-	case HL_PRSTN_RST_SRC:
+	case HL_PRSTN_RST:
 		cause = "prstn";
 		break;
-	case HL_WD_RST_SRC:
+	case HL_WD_RST:
 		cause = "watch dog";
 		break;
-	case HL_SW_ALL_RST_SRC:
+	case HL_SW_ALL_RST:
 		cause = "software all";
 		break;
 	default:
@@ -164,7 +164,7 @@
 
 	/* check source of reset */
 	rst_src = hl_rst_src_get();
-	if ((rst_src == HL_COLD_RST_SRC) || (rst_src == HL_PRSTN_RST_SRC))
+	if ((rst_src == HL_COLD_RST) || (rst_src == HL_PRSTN_RST))
 		return false;
 
 	prstn = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN);
diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 5b72dad..ccbc313 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -343,7 +343,7 @@
 	 * required to clear previous indication.
 	 */
 	rst_src = hl_rst_src_get();
-	if (rst_src != HL_PRSTN_RST_SRC)
+	if (rst_src != HL_PRSTN_RST)
 		boot_info_get(HL_BOOT_INFO_FW_UPD_CLOSURE, &fw_upd_is_dis,
 			      sizeof(fw_upd_is_dis), &data_size);
 
diff --git a/arch/gaudi/arm/asic.c b/arch/gaudi/arm/asic.c
index 326468a..c5462e6 100644
--- a/arch/gaudi/arm/asic.c
+++ b/arch/gaudi/arm/asic.c
@@ -277,7 +277,7 @@
 	u32 rst_src;
 
 	rst_src = hl_rst_src_get();
-	return rst_src == HL_WD_RST_SRC;
+	return rst_src == HL_WD_RST;
 }
 
 void preboot_finalize_fw_info(void)
diff --git a/arch/gaudi/arm/hl.h b/arch/gaudi/arm/hl.h
index df7eff3..85aaba6 100644
--- a/arch/gaudi/arm/hl.h
+++ b/arch/gaudi/arm/hl.h
@@ -7,11 +7,11 @@
 #define ARCH_GAUDI_ARM_HL_H_
 
 enum hl_rst_src {
-	HL_COLD_RST_SRC,
-	HL_MANUAL_RST_SRC,
-	HL_PRSTN_RST_SRC = 2,
-	HL_WD_RST_SRC = 4,
-	HL_SW_ALL_RST_SRC = 8,
+	HL_COLD_RST,
+	HL_MANUAL_RST,
+	HL_PRSTN_RST = 2,
+	HL_WD_RST = 4,
+	HL_SW_ALL_RST = 8,
 };
 
 #endif /* ARCH_GAUDI_ARM_HL_H_ */
diff --git a/arch/gaudi/arm/pll.c b/arch/gaudi/arm/pll.c
index 09a4eaf..6962334 100644
--- a/arch/gaudi/arm/pll.c
+++ b/arch/gaudi/arm/pll.c
@@ -138,7 +138,7 @@
 
 	/* execute pci pll WA only when required */
 	rst_src = hl_rst_src_get();
-	if ((rst_src == HL_COLD_RST_SRC) || (rst_src == HL_PRSTN_RST_SRC)) {
+	if ((rst_src == HL_COLD_RST) || (rst_src == HL_PRSTN_RST)) {
 		for (i = 0; i < ARRAY_SIZE(gaudi_pll_cfg); i++) {
 			if (strcmp(gaudi_pll_cfg[i].name, "PCI") != 0)
 				continue;
diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index c5f3551..cc2d753 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -52,7 +52,7 @@
 	u32 rst_src = hl_rst_src_get();
 
 	/* If not cold reset, dont download PHY FW */
-	if (!(rst_src & BIT(HL_COLD_RST_SRC)))
+	if (!(rst_src & BIT(HL_COLD_RST)))
 		return;
 
 	for (i = 0; i < (size / 4); i++)
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 8327be6..2d60d83 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -717,31 +717,31 @@
 	const char *cause;
 
 	switch (rst_src) {
-	case HL_COLD_RST_SRC:
+	case HL_COLD_RST:
 		cause = "cold";
 		break;
-	case HL_MANUAL_RST_SRC:
+	case HL_MANUAL_RST:
 		cause = "manual";
 		break;
-	case HL_PRSTN_RST_SRC:
+	case HL_PRSTN_RST:
 		cause = "prstn";
 		break;
-	case HL_SOFT_RST_SRC:
+	case HL_SOFT_RST:
 		cause = "soft";
 		break;
-	case HL_WD_RST_SRC:
+	case HL_WD_RST:
 		cause = "watch dog";
 		break;
-	case HL_FW_ALL_RST_SRC:
+	case HL_FW_ALL_RST:
 		cause = "firmware all";
 		break;
-	case HL_SW_ALL_RST_SRC:
+	case HL_SW_ALL_RST:
 		cause = "software all";
 		break;
-	case HL_FLR_RST_SRC:
+	case HL_FLR_RST:
 		cause = "functional";
 		break;
-	case HL_ECC_DERR_RST_SRC:
+	case HL_ECC_DERR_RST:
 		cause = "ecc derr";
 		break;
 	default:
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index 9f5e759..5ef6af3 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -6,19 +6,8 @@
 #ifndef ARCH_GAUDI2_COMMON_HL_H_
 #define ARCH_GAUDI2_COMMON_HL_H_
 
+#include <gaudi2/gaudi2_fw_if.h>
 #include "common.h"
-
-enum hl_rst_src {
-	HL_COLD_RST_SRC = 1,
-	HL_MANUAL_RST_SRC = 2,
-	HL_PRSTN_RST_SRC = 4,
-	HL_SOFT_RST_SRC = 8,
-	HL_WD_RST_SRC = 16,
-	HL_FW_ALL_RST_SRC = 32,
-	HL_SW_ALL_RST_SRC = 64,
-	HL_FLR_RST_SRC = 128,
-	HL_ECC_DERR_RST_SRC = 256
-};
 
 void set_pci_msix_pba_offset_val(void);
 void set_slv_cfg_reg_val(void);
diff --git a/arch/goya/arm/hl.h b/arch/goya/arm/hl.h
index bd581e8..6c5c535 100644
--- a/arch/goya/arm/hl.h
+++ b/arch/goya/arm/hl.h
@@ -7,11 +7,11 @@
 #define ARCH_GOYA_ARM_HL_H_
 
 enum hl_rst_src {
-	HL_COLD_RST_SRC,
-	HL_MANUAL_RST_SRC,
-	HL_PRSTN_RST_SRC = 2,
-	HL_WD_RST_SRC = 4,
-	HL_SW_ALL_RST_SRC = 8,
+	HL_COLD_RST,
+	HL_MANUAL_RST,
+	HL_PRSTN_RST = 2,
+	HL_WD_RST = 4,
+	HL_SW_ALL_RST = 8,
 };
 
 #endif /* ARCH_GOYA_ARM_HL_H_ */
diff --git a/arch/greco/arc_elf32/asic.c b/arch/greco/arc_elf32/asic.c
index d9b1928..3a6a39a 100644
--- a/arch/greco/arc_elf32/asic.c
+++ b/arch/greco/arc_elf32/asic.c
@@ -36,7 +36,7 @@
 	u32 *dst = (u32 *)PCIE_FW_SRAM_BASE;
 	u32 rst_src = hl_rst_src_get();
 
-	if (rst_src != HL_COLD_RST_SRC)
+	if (rst_src != HL_COLD_RST)
 		return;
 
 	for (i = 0; i < (size / 4); i++)
diff --git a/arch/greco/common/hl.h b/arch/greco/common/hl.h
index ab7f5d7..e441440 100644
--- a/arch/greco/common/hl.h
+++ b/arch/greco/common/hl.h
@@ -41,11 +41,11 @@
 #define ARC1_MEM_SIZE			(ARC1_ICCM_SIZE + ARC1_DCCM_SIZE)
 
 enum hl_rst_src {
-	HL_COLD_RST_SRC,
-	HL_MANUAL_RST_SRC,
-	HL_PRSTN_RST_SRC = 2,
-	HL_WD_RST_SRC = 4,
-	HL_SW_ALL_RST_SRC = 8,
+	HL_COLD_RST,
+	HL_MANUAL_RST,
+	HL_PRSTN_RST = 2,
+	HL_WD_RST = 4,
+	HL_SW_ALL_RST = 8,
 };
 
 void run_arc(u8 id);
diff --git a/lib/preboot/module_pci.c b/lib/preboot/module_pci.c
index 667d88e..6cf219c 100644
--- a/lib/preboot/module_pci.c
+++ b/lib/preboot/module_pci.c
@@ -131,7 +131,7 @@
 	unsigned int rst_src = hl_rst_src_get();
 
 	if (!pci_sts.pci_pll_initialized) {
-		if ((rst_src == HL_COLD_RST_SRC) || (rst_src == HL_PRSTN_RST_SRC))
+		if ((rst_src == HL_COLD_RST) || (rst_src == HL_PRSTN_RST))
 			hl_dev_pci_pll_wa();
 
 		pci_sts.pci_pll_initialized = true;
@@ -167,8 +167,8 @@
 	if (pci_sts.mem_enabled_prev != pci_sts.mem_enabled) {
 		pci_sts.mem_enabled_prev = pci_sts.mem_enabled;
 		rst_src = hl_rst_src_get();
-		if (pci_sts.mem_enabled && ((rst_src == HL_COLD_RST_SRC) ||
-					    (rst_src == HL_PRSTN_RST_SRC)))
+		if (pci_sts.mem_enabled && ((rst_src == HL_COLD_RST) ||
+					    (rst_src == HL_PRSTN_RST)))
 			hl_pci_iatu_init();
 
 		preboot_puts_raw("PCIe mem_en changed to: ");
